### PR TITLE
Goroutine wrapper, version print format fix, stats client creation re…

### PR DESCRIPTION
**Pull request by chandumlg**
…structure

### CODE REVIEW

File: services/stats/stats.go [hunk 1]
```diff
@@ -47,17 +47,23 @@ func init() {
 	enableCPUStats = config.GetBool("RuntimeStats.enableCPUStats", true)
 	enableMemStats = config.GetBool("RuntimeStats.enabledMemStats", true)
 	enableGCStats = config.GetBool("RuntimeStats.enableGCStats", true)
+}
+
+//CreateStatsClient creates a new statsd client
+func CreateStatsClient() {
```
Comment (snarkychef) [inline services/stats/stats.go]: Let's call it something like initialize or setup?

File: rruntime/goroutine-factory.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	"runtime"
+	"runtime/debug"
+
+	"github.com/bugsnag/bugsnag-go"
+	"github.com/rudderlabs/rudder-server/utils/logger"
+	"github.com/rudderlabs/rudder-server/utils/misc"
+)
+
+//StartGoroutine Starts the excution of the function passed as argument in a new Goroutine
+func StartGoroutine(function func()) {
```
[the remark this answers is not in the document]
Comment (snarkychef) [reply rruntime/goroutine-factory.go]: There is a problem with this pattern. See and execute following code that highlights the issue:

`package main

import "fmt"
import "time"

func main() {
	str := "hellow"
	defer fmt.Println(str)
	startGoRoutine(func(){
             test(str)
         })
	str = "check"
	fmt.Println(str)
	time.Sleep(time.Second * 5)
}

func startGoRoutine(function func()){
   go func(){
   //time.Sleep(time.Second * 2)
   function()
   }()
}

func test(str string) {
    time.Sleep(time.Second * 2)
	fmt.Println("inside test", str)
} `

File: config/backend-config/backend-config.go [hunk 3]
```diff
@@ -15,6 +15,7 @@ import (
 	"github.com/rudderlabs/rudder-server/utils/logger"
 
 	"github.com/rudderlabs/rudder-server/config"
+	GoroutineFactory "github.com/rudderlabs/rudder-server/rruntime"
```
[the remark this answers is not in the document]
Comment (chandumlg) [reply config/backend-config/backend-config.go]: Removed alias.

File: config/backend-config/multi-workspace-config.go [hunk 4]
```diff
@@ -26,7 +26,7 @@ func (multiWorkspaceConfig *MultiWorkspaceConfig) SetUp() {
 	multiWorkspaceConfig.writeKeyToWorkspaceIDMap = make(map[string]string)
 }
 
-//GetWorkspaceIDForWriteKey return workspaceID for the given writeKey
+//GetWorkspaceIDForWriteKey returns workspaceID for the given writeKey
 func (multiWorkspaceConfig *MultiWorkspaceConfig) GetWorkspaceIDForWriteKey(givenWriteKey string) string {
```
[the remark this answers is not in the document]
Comment (chandumlg) [reply config/backend-config/multi-workspace-config.go]: Done

File: rruntime/goroutine-factory.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (chandumlg) [reply rruntime/goroutine-factory.go]: Written a note in the comment regarding the usage of the function.

File: utils/misc/misc.go [hunk 5]
```diff
@@ -115,11 +112,6 @@ func RecordAppError(err error) {
 //AssertError panics if error
 func AssertError(err error) {
```
[the remark this answers is not in the document]
Comment (chandumlg) [reply utils/misc/misc.go]: Done.

File: utils/misc/misc.go [hunk 6]
```diff
@@ -133,23 +125,13 @@ func AssertErrorIfDev(err error) {
...
-		RecordAppError(err)
-		logger.Fatal(err)
 		panic(err)
 	}
 }
 
 //Assert panics if false
 func Assert(cond bool) {
 	if !cond {
-		//debug.SetTraceback("all")
```
[the remark this answers is not in the document]
Comment (chandumlg) [reply utils/misc/misc.go]: Done.

File: services/stats/stats.go (hunk 1 above)
Comment (chandumlg) [reply services/stats/stats.go]: Done

File: jobsdb/jobsdb.go [hunk 7]
```diff
@@ -169,13 +163,8 @@ func (jd *HandleT) assertErrorIfDev(err error) {
 
 func (jd *HandleT) assert(cond bool) {
```
[the remark this answers is not in the document]
Comment (chandumlg) [reply jobsdb/jobsdb.go]: Done

**Pull request merged**